src/alchemlyb/tests/parsing/test_gmx.py

```diff
@@ -124,7 +124,7 @@ def test_u_nk_with_total_energy():
 
     # Check one specific value in the dataframe
     assert_almost_equal(
-        extract_u_nk(dataset['data']['AllStates'][0], T=300).iloc[0][0],
+        extract_u_nk(dataset['data']['AllStates'][0], T=300).loc[0][(0.0,0.0)].values[0],
```

Comment:
PEP8 and using `.loc`
```suggestion
      extract_u_nk(dataset['data']['AllStates'][0], T=300).loc[0, (0.0, 0.0)].values[0],
```